beacon-chain/blockchain/service.go

```diff
@@ -299,11 +274,8 @@ func (c *ChainService) blockProcessing() {
 
 			log.Infof("Finished processing received block: %#x", blockHash)
 
-			// We push the hash of the block we just stored to a pending processing
-			// slice the fork choice rule will utilize.
-			c.lock.Lock()
-			c.blocksPendingProcessing = append(c.blocksPendingProcessing, blockHash)
-			c.lock.Unlock()
+			// Push the block to trigger the fork choice rule
+			processedBlock <- block
```

Comment:
What if we get tons of blocks within a slot interval? will we be calculating GHOST every single time?

Reply:
I was concerned about that too, but in production we can expect only one valid block per slot. Multiple blocks for the same slot is a slashing condition, and the validator's deposit is going to be taken away.

It gets tricky when we're running the simulator and a validator node at the same time. This would trigger a slashing condition in prod, but at the moment it just results in `updateHead` getting called twice.

Reply:
>Multiple blocks for the same slot is a slashing condition

Attesting to multiple blocks for the same slot is a slashing condition. This doesn't prevent receiving multiple blocks at the same slot though

Reply:
It's a slashing condition in the spec under `NO_DBL_PROPOSE`.